airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py

```diff
@@ -162,3 +203,34 @@ def get_custom_object_streams(self, api: API, common_params: Mapping[str, Any]):
                 custom_properties=custom_properties,
                 **common_params,
             )
+
+    def get_web_analytics_custom_objects_stream(
+        self, custom_object_stream_instances: List[CustomObject], common_params: Any
+    ) -> WebAnalyticsStream:
+        for custom_object_stream_instance in custom_object_stream_instances:
+
+            def __init__(self, **kwargs: Any):
+                parent = custom_object_stream_instance.__class__(
+                    entity=custom_object_stream_instance.entity,
+                    schema=custom_object_stream_instance.schema,
+                    fully_qualified_name=custom_object_stream_instance.fully_qualified_name,
+                    custom_properties=custom_object_stream_instance.custom_properties,
+                    **common_params,
+                )
+                super(self.__class__, self).__init__(parent=parent, **kwargs)
+
+            def get_json_schema(self):
```

Comment:
We can but there is one proper place where we define schemas per stream. I have already made one common schema for those streams that share same properties. For now this schema is only one source of fields because I can't get fields specific for every event type, but I think it will be possible in the future, so we will extend schema related to particular stream. Current method seems suitable for now for all the streams but I think it is an exception and this is only a place good for its definition